.github/workflows/build-test-e2e.yml

```diff
@@ -0,0 +1,52 @@
+name: 'Build/Run End-to-End Tests'
+
+on: push
+jobs:
+  build-and-run-e2e:
+    if: ${{ github.ref == 'develop' || github.ref == 'release-**' || github.actor == github.repository_owner }}
```

Comment:
Repo secrets can't be accessed in workflows triggered by pull requests from forked repos, as a security measure to prevent exfiltrating secrets. A workflow can access secrets if the `pull_request_target` trigger is used, but would use the base (develop) branch workflow and code, rather than the PR code. Since the e2e tests wouldn't be run against the PR code, this wouldn't be very useful for our use case. So, we will only run end-to-end tests on push to `develop` and `release-*` branches in the `Sage-Bionetworks` repo.

However, since tests that pass locally can be flaky in CI, we would like devs to know the status of e2e tests before merging a PR to develop. So, also run the e2e tests whenever a dev pushes changes to their forked repo. The workflow will be able to access secrets stored within the forked repo, since they are the repo owner.